### ISSUE TITLE
Delve debugger variable loading limits too small

### ISSUE THREAD

**Issue by defrancr** [golang/vscode-go#2733]
Hi there,

The current limit of 64 chars in `dlv-dap` is too small. IMO, a user should be able to configure it, but if that is hard to achieve, a rise to 128 or 256 will help. 

Thank you.

**Comment by hyangah** [comment]
Where do you see 64char limit?
According to the doc 
https://github.com/golang/vscode-go/wiki/debugging#i-need-to-view-large-strings-how-can-i-do-that-if-dlvloadconfig-with-maxstringlen-is-deprecated
the current limit is 512.

**Comment by defrancr** [comment]
@hyangah, you are right, it is not reproduced anymore. Thank you and sorry for the inconvenience.

**Comment by defrancr** [comment]
Sorry @hyangah, it is still reproducible for byte types, for example. Please see attached image.

![image](https://github.com/golang/vscode-go/assets/21229843/29420f33-311e-4525-ab31-1b418a8c5e34)

**Comment by hyangah** [comment]
Can you try to expand the "example" variable? 
You can access the value with `string()` which will allow larger limit like other string vars or 
you can access individual bytes by accessing the index.  
![Screenshot 2023-05-15 at 5 44 16 PM](https://github.com/golang/vscode-go/assets/4999471/ab871b78-3f5d-4024-9e40-4f70840f14e7)

If that doesn't meet your need, we can consider to add a similar trick for `[]byte`. 
(Alternative is to increase [`MaxArrayValues`](https://github.com/go-delve/delve/blob/463b97dd3661f5f74b6ad17116a00ff893faf2d5/service/dap/server.go#L261) but I think applying a larger limit to all array types can slow down debugging significantly - @suzmue )

**Comment by defrancr** [comment]
Hi @hyangah, sorry for the late answer. Applying string() definitively works for those bytes with "String" representation. IMO, the best solution would be to have this limit configurable by the user in settings.json file, like `"go.maxArrayValues": 64`, so every user can be responsible of their decisions (sometimes it could make sense to slow down debugging)

**Comment by cnp-lab** [comment]
Hi, it's weird but I got still problem with displaying longer strings - my max limit is 64 chars.
I use the newest VSCode 1.80.0 and the newest vscode.go plugin v0.39.0.

![image](https://github.com/golang/vscode-go/assets/127735285/bf608757-a1d3-456d-b700-dce0bfd4c875)


Do you have any tips what I can do ?

**Comment by hyangah** [comment]
@cnp-lab Thanks for the screenshot. Are you using remote debugging ? In case, can you try if switching to the delve dap helps?   If you have `launch.json`, adding `"debugAdapter": "dlv-dap"` will make your debugging session to use delve dap. Details:  https://github.com/golang/vscode-go/wiki/debugging#remote-debugging

**Comment by natenho** [comment]
VSCode could have the string visualizer like the good old Visual Studio:

https://learn.microsoft.com/en-us/visualstudio/debugger/view-strings-visualizer?view=vs-2022
![image](https://github.com/golang/vscode-go/assets/4236481/cb3dc7f2-849a-425b-b55b-fc8cbe22b0ac)

And that should work for strings, []byte and so on.

There are some cases where we need to see or copy the full string, for instance a full HTML document or a relatively big JSON variable passing through an application.

Please reconsider this limitation or at least provide the option to increase like the legac dlv :(

**Comment by gudvinr** [comment]
Even with explicit `"debugAdapter": "dlv-dap"` in launch.json "Copy Value" copies 512 characters and preview shows only  256 characters.

<img width="642" height="193" alt="Image" src="https://github.com/user-attachments/assets/faea418b-07ce-406a-9ab8-8df92741650c" />

That is _after_ using Copy Value

**Comment by firelizzard18** [comment]
This is a limitation of delve, not vscode-go. Once google/go-dap#94 and go-delve/delve#4084 are merged, golang/vscode-go#3818 will provide a workaround.
